modules/localstack/src/test/java/org/testcontainers/containers/localstack/LocalstackContainerTest.java

```diff
@@ -0,0 +1,144 @@
+package org.testcontainers.containers.localstack;
+
+
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3ClientBuilder;
+import com.amazonaws.services.s3.model.Bucket;
+import com.amazonaws.services.s3.model.ObjectListing;
+import com.amazonaws.services.s3.model.S3Object;
+import com.amazonaws.services.sqs.AmazonSQS;
+import com.amazonaws.services.sqs.AmazonSQSClientBuilder;
+import com.amazonaws.services.sqs.model.CreateQueueResult;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.io.IOUtils;
+import org.junit.Assert;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.testcontainers.DockerClientFactory;
+import org.testcontainers.containers.Container;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.Network;
+
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.util.List;
+
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.rnorth.visibleassertions.VisibleAssertions.assertEquals;
+import static org.rnorth.visibleassertions.VisibleAssertions.assertThat;
+import static org.rnorth.visibleassertions.VisibleAssertions.assertTrue;
+import static org.testcontainers.containers.localstack.LocalStackContainer.Service.S3;
+import static org.testcontainers.containers.localstack.LocalStackContainer.Service.SQS;
+
+/**
+ * Tests for Localstack Container, used both in bridge network (exposed to host) and docker network modes.
+ *
+ * These tests attempt simple interactions with the container to verify behaviour. The bridge network tests use the
+ * Java AWS SDK, whereas the docker network tests use an AWS CLI container within the network, to simulate usage of
+ * Localstack from within a Docker network.
+ */
+@Slf4j
+public class LocalstackContainerTest {
+
+    // without_network {
+    @ClassRule
```

Comment:
Can we please at least use nested test classes if we use the rules?
Starting 3 localstack containers at once may be very slow in some CIs

Reply:
Shouldn't we even consider making this a plain test without rules? 
Also considering our way forward, which would put rules into another module.

Reply:
I would very much be in favour of it :)

Reply:
Very good point about the weight of these containers.

TBH I think I prefer the nested test class approach here; it's not a situation where try-with-resources is particularly elegant, and because these tests are excerpted for docs I'd like to keep the container creation style consistent.

I'll push a change with nested classes in a moment.